src/movies.js

```diff
@@ -1,22 +1,2081 @@
+      genre: ['Crime', 'Drama', 'Mystery', 'Romance', 'Thriller'],
+      score: 8.1
+    },
+    {
+      title: 'Piscores of the Caribbean: The Curse of the Black Pearl',
+      year: 2003,
+      director: 'Gore Verbinski',
+      duration: '2h 23min',
+      genre: ['Action', 'Adventure', 'Fantasy'],
+      score: 8
+    },
+    {
+      title: 'PK',
+      year: 2014,
+      director: 'Rajkumar Hirani',
+      duration: '2h 33min',
+      genre: ['Comedy', 'Drama', 'Fantasy', 'Sci-Fi'],
+      score: 8.2
+    },
+    {
+      title: 'Dog Day Afternoon',
+      year: 1975,
+      director: 'Sidney Lumet',
+      duration: '2h 5min',
+      genre: ['Biography', 'Crime', 'Drama', 'Thriller'],
+      score: 8
+    },
+    {
+      title: 'Dead Poets Society',
+      year: 1989,
+      director: 'Peter Weir',
+      duration: '2h 8min',
+      genre: ['Comedy', 'Drama'],
+      score: 8
+    }
+  ];
+  
+    
+function getAllDirectors(moviesArray) {
+   const directorsArray = movies.map((movie) => movie.director)
+    return directorsArray ;}
+
+// console.log(getAllDirectors(movies))
+
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+    const spielbergMovies = movies.filter((movie) => movie.director == 'Steven Spielberg')
+    return `Steven Spielberg directed ${spielbergMovies.length} movies`
+}
+// console.log(howManyMovies(movies))
+
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+
+function scoresAverage(moviesArray) {
+    const averageScore = moviesArray.reduce((sum, movie) => 
+              sum + movie.score, 0)
+              return (averageScore / moviesArray.length).toFixed(2)
+}
+    //console.log(scoresAverage(movies))
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore(moviesArray) {}
+function dramaMoviesScore(moviesArray) {  
+    const dramaMovies = moviesArray.filter((movie) => movie.genre == 'Drama')
+    const averageDramaScore = dramaMovies.reduce((sum, movie) => 
+    sum + movie.score, 0)
+    return (averageDramaScore / dramaMovies.length).toFixed(2)
+}
+//console.log(dramaMoviesScore(movies))
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear(moviesArray) {}
+function orderByYear(moviesArray) {
+const orderedMovies = moviesArray.slice()
+
+function compare(a, b){
+if(a.year < b.year){
+    return -1;}
+
+if(a.year > b.year){
+    return 1;
+}
+else (a.title < b.title)
+    return 1
+}
+return(orderedMovies.sort(compare))
+
+}
+ console.log(orderByYear(movies)) 
 
 // Iteration 6: Alphabetic Order - Order by title and print the first 20 titles
-function orderAlphabetically(moviesArray) {}
+function orderAlphabetically(moviesArray) {
+    const orderedMovies = moviesArray.slice()
+
+    function compare(a, b){
```

Comment:
To clone the original array and keep just the titles, you could do this:
```
const onlyTitles = moviesArray.map((movies) => movies.title);
```
After you clone the original array and get just the titles, you could use the `.sort()` method: 
```
onlyTitles.sort(function (a, b) {
    return a.localeCompare(b);
  });
  return onlyTitles.slice(0, 20);
```